src/Microsoft.Tye.Core/ConfigModel/ConfigIngress.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.Tye.ConfigModel
     public class ConfigIngress
     {
         [Required]
-        public string Name { get; set; } = default!;
+        public string? Name { get; set; } = default!;
```

Comment:
Remove `default!` if you're going to make these nullable. 

The point of how this worked in the past:

The nullability contract + `[Required]` reflect the **valid** state of the object. That's what `default!` is used. If you're going to treat the properties as nullable, there's no need to initialized them explicitly.